pkg/find/vulnerabilities/scanner.go

```diff
@@ -25,39 +24,45 @@ import (
 	"k8s.io/utils/pointer"
 )
 
+// Scanner is a template for running static vulnerability scanners that implement
+// the vulnerabilityreport.Plugin interface.
 type Scanner struct {
-	scheme      *runtime.Scheme
-	config      starboard.TrivyConfig
-	opts        kube.ScannerOpts
-	clientset   kubernetes.Interface
-	pods        *pod.Manager
-	converter   trivy.Converter
-	idGenerator ext.IDGenerator
-	delegate    vulnerabilityreport.Plugin
+	scheme    *runtime.Scheme
+	clientset kubernetes.Interface
+	opts      kube.ScannerOpts
+	pods      *pod.Manager
+	ext.IDGenerator
+	vulnerabilityreport.Plugin
 	kube.SecretsReader
 }
 
-// NewScanner constructs a new vulnerability Scanner with the specified options and Kubernetes client Interface.
+// NewScanner constructs a new static vulnerability Scanner with the specified
+// vulnerabilityreport.Plugin that knows how to perform the actual scanning,
+// which is performed by running a Kubernetes job, and knows how to convert logs
+// to instances of v1alpha1.VulnerabilityReport.
```

Comment:
I wonder if this eventually becomes an even more generic Scanner that is configured to run a particular type of scan, and convert to a configurable type of report? Doesn't need to happen now in any case, I'm just wondering about it

Reply:
All scanners look similar in terms of creating a job, waiting for its completion, parsing logs, and finally writing report to etcd. So yes, maybe one day we can generalize this code as you said.